File: apps/api/src/main/java/app/weather/dto/WeatherSummaryResponse.java
package app.weather.dto;

import java.util.List;

public record WeatherSummaryResponse(
    String locationName,
    String regionName,
    double latitude,
    double longitude,
    double currentTemp,
    double highTemp,
    double lowTemp,
    double feelsLikeTemp,
    int humidityPercent,
    int precipitationChance,
    int windSpeedKph,
    String description,
    String conditionLabel,
    String source,
    boolean cached,
    int airQualityIndex,
    String airQualityLabel,
    String airQualitySummary,
    String updatedAt,
    List<HourlyForecastPoint> hourly,
    List<DailyForecastPoint> daily
) {}

record HourlyForecastPoint(
    String hourLabel,
    double temperature,
    int precipitationChance,
    int windSpeedKph,
    String icon
) {}

record DailyForecastPoint(
    String dayLabel,
    String dateLabel,
    double highTemp,
    double lowTemp,
    int precipitationChance,
    String icon
) {}

File: apps/api/src/main/kotlin/app/weather/model/WeatherSummaryResponse.kt
package app.weather.model

data class WeatherSummaryResponse(
    val locationName: String,
    val regionName: String,
    val latitude: Double,
    val longitude: Double,
    val currentTemp: Double,
    val highTemp: Double,
    val lowTemp: Double,
    val feelsLikeTemp: Double,
    val humidityPercent: Int,
    val precipitationChance: Int,
    val windSpeedKph: Int,
    val description: String,
    val conditionLabel: String,
    val source: String,
    val cached: Boolean,
    val airQualityIndex: Int,
    val airQualityLabel: String,
    val airQualitySummary: String,
    val updatedAt: String,
    val hourly: List<HourlyForecastPoint>,
    val daily: List<DailyForecastPoint>,
)

data class HourlyForecastPoint(
    val hourLabel: String,
    val temperature: Double,
    val precipitationChance: Int,
    val windSpeedKph: Int,
    val icon: String,
)

data class DailyForecastPoint(
    val dayLabel: String,
    val dateLabel: String,
    val highTemp: Double,
    val lowTemp: Double,
    val precipitationChance: Int,
    val icon: String,
)
